lib/activerecord-multi-tenant/query_monitor.rb

```diff
@@ -6,11 +6,34 @@ def self.enable_query_monitor; @@enable_query_monitor = true; end
   def self.query_monitor_enabled?; @@enable_query_monitor; end
 
   class QueryMonitor
-    def start(name, id, payload); end
-    def finish(name, id, payload)
+    def finish(_name, _id, _payload); end
+
+    def start(_name, _id, payload)
       return unless MultiTenant.query_monitor_enabled?
-      return unless payload[:exception].present? && MultiTenant.current_tenant_id.nil?
-      Rails.logger.info 'WARNING: Tenant not present - make sure to add MultiTenant.with(tenant) { ... }'
+      return unless MultiTenant.current_tenant_id.nil?
+      return if MultiTenant.explicit_without
+      return if call_part_of_setup?(payload[:sql])
+
+      message = 'WARNING: Tenant not present - make sure to add MultiTenant.with(tenant) {...}'
+      ActiveRecord::Base.logger.warn "#{message} #{payload[:sql]}"
+    end
+
+    private
+
+    def call_part_of_setup?(trace)
+      call_to_set_or_show(trace) || call_to_pg_tables(trace)
+    end
+
+    def call_to_set_or_show(trace)
+      trace =~ /^SET/ || trace =~ /^SHOW/
```

Comment:
Might it be slightly cheaper to combine these, in case the first match fails?

```suggestion
      trace =~ /^(?:SET|SHOW)/
```